src/api/localResolvers/loan.js

```diff
@@ -0,0 +1,166 @@
+ *
+ * Example fragment:
+ * ... on LoanBasic {
+	 loanAmount
+	 loanFundraisingInfo {
+		 fundedAmount
+		 reservedAmount
+	 }
+	 fundraisingPercent @client
+ * }
+* */
+function fundraisingPercent(loan) {
+	// Handle missing required loan fields
+	if (missingLoanFields({
+		loan,
+		fields: [
+			'loanAmount',
+			'loanFundraisingInfo.fundedAmount',
+			'loanFundraisingInfo.reservedAmount'
+		],
+		resolver: 'fundraisingPercent',
+	})) {
+		return 0;
+	}
+
+	// Get amounts as numbers
+	const loanAmount = numeral(loan.loanAmount).value();
+	const fundedAmount = numeral(loan.loanFundraisingInfo.fundedAmount).value();
+	const reservedAmount = numeral(loan.loanFundraisingInfo.reservedAmount).value();
+
+	// Calculate progress percentage
+	const progressPercent = (fundedAmount + reservedAmount) / loanAmount;
+
+	// Ensure returned value is not less than 0 and is not greater than 1
+	return Math.max(0, Math.min(1, progressPercent));
+}
+
+/**
+ * Resolves the time left before expiration as words.
+ *
+ * Example fragment:
+ * ... on LoanBasic {
+	 plannedExpirationDate
+	 fundraisingTimeLeft @client
+ * }
+* */
+function fundraisingTimeLeft(loan) {
+	// Handle missing required field
+	if (missingLoanField({
+		loan,
+		field: 'plannedExpirationDate',
+		resolver: 'fundraisingTimeLeft',
+	})) {
+		return '';
+	}
+
+	// Get planned expiration time as Date
+	const plannedExpirationDate = parseISO(loan.plannedExpirationDate);
+
+	// Return empty string if the result is not a valid date
+	if (!isValidDate(plannedExpirationDate)) {
+		return '';
+	}
+
+	// Return the distance between the expiration time and now in words.
+	return formatDistanceToNowStrict(plannedExpirationDate);
+}
+
+/**
+ * Resolves the unreserved amount of this loan as a Money string.
+ *
+ * Example fragment:
+ * ... on LoanBasic {
+	 loanAmount
+	 loanFundraisingInfo {
+		 fundedAmount
+		 reservedAmount
+	 }
+	 unreservedAmount @client
+ * }
+* */
+function unreservedAmount(loan) {
+	// Handle missing required loan fields
+	if (missingLoanFields({
+		loan,
+		fields: [
+			'loanAmount',
+			'loanFundraisingInfo.fundedAmount',
+			'loanFundraisingInfo.reservedAmount'
+		],
+		resolver: 'unreservedAmount',
+	})) {
+		return '0.00';
+	}
+
+	// Get amounts as numbers
+	const loanAmount = numeral(loan.loanAmount).value();
+	const fundedAmount = numeral(loan.loanFundraisingInfo.fundedAmount).value();
+	const reservedAmount = numeral(loan.loanFundraisingInfo.reservedAmount).value();
+
+	// Calculate unfunded and unreserved amounts
+	const unfunded = loanAmount - fundedAmount;
+	const unreserved = unfunded - reservedAmount;
+
+	// Format as string to match Money type
+	return numeral(unreserved).format('0.00');
+}
+
+export default () => {
+	return {
+		resolvers: {
+			LoanPartner: {
+				fundraisingPercent,
+				fundraisingTimeLeft,
+				unreservedAmount,
+			},
+			LoanDirect: {
+				fundraisingPercent,
+				fundraisingTimeLeft,
+				unreservedAmount,
+			},
```

Comment:
This is coming together nicely. I'm assuming these local resolver fields will expand with the page so it would be nice  move the plain functions that power them out into a loan util file at some point.